scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/view/grid/jobs/JobsListGrid.java

```diff
@@ -488,17 +487,58 @@ protected void buildCellContextualMenu(Menu menu) {
         removeItem.addClickHandler(event -> controller.removeJob(ids));
         removeItem.setEnabled(selFinished);
 
-        menu.setItems(pauseItem,
-                      restartInErrorTaskItem,
-                      resumeItem,
-                      resumeAndRestartItemTask,
-                      priorityItem,
-                      killItem,
-                      killAndResubmitItem,
-                      resubmitItem,
-                      openItem,
-                      exportXmlItem,
-                      removeItem);
+        SchedulerServiceAsync scheduler = Scheduler.getSchedulerService();
+        boolean finalSelPauseOrRunning = selPauseOrRunning;
+        scheduler.getJobInfoDetails(LoginModel.getInstance().getSessionId(), ids.get(0), new AsyncCallback<String>() {
+            public void onSuccess(String result) {
```

Comment:
This code has to be in the controller not in the view

Reply:
I did like this in the begining but I needed to get the signals before creating the Action menu and since the callback is asynchronous I had to create the menu when I receive the signals (on onSuccess method) so there would be no delay

Reply:
you can call a method from the controller + pass the needed widgets as parameters to the controller method